File: app/src/main/java/tw/ctl/interest/Entity.java
package tw.ctl.interest;

import java.util.Date;

import io.realm.Realm;
import io.realm.RealmObject;

/**
 * Created by jacklai on 2017/3/1.
 */

public class Entity extends RealmObject {

    public String principal;
    public String interest;
    public String period;
    public String invest;

    public String simpleResult;
    public String compoundResult;
    public String investResult;

    public Date date;

    public Entity() {}

    public void save() {
        date = new Date();
        Realm realm = Realm.getDefaultInstance();
        realm.beginTransaction();
        realm.copyToRealm(this);
        realm.commitTransaction();
        realm.close();
    }

    public void delete() {
        if (!this.isValid()) return;
        Realm realm = Realm.getDefaultInstance();
        realm.beginTransaction();
        deleteFromRealm();
        realm.commitTransaction();
        realm.close();
    }

}

File: app/src/main/java/tw/ctl/interest/Entity.kt
package tw.ctl.interest

import io.realm.Realm
import io.realm.RealmObject
import java.util.*

open class Entity(
        var principal: String = "",
        var interest: String = "",
        var period: String = "",
        var invest: String = "",
        var simpleResult: String = "",
        var compoundResult: String = "",
        var investResult: String = "",
        private var date: Date? = null
) : RealmObject() {

    fun save() {
        date = Date()
        val realm = Realm.getDefaultInstance()
        realm.beginTransaction()
        realm.copyToRealm(this)
        realm.commitTransaction()
        realm.close()
    }

    fun delete() {
        if (!this.isValid) return
        val realm = Realm.getDefaultInstance()
        realm.beginTransaction()
        deleteFromRealm()
        realm.commitTransaction()
        realm.close()
    }

}